### PR TITLE
Add tables-io package

### PR DESCRIPTION
**Pull request by brianv0**
Checklist
- [X] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [X] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example).
- [X] Source is from official source.
- [X] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [X] Build number is 0.
- [X] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by brianv0** [comment]
@conda-forge/help-python ready

osx is still going (slowly) for some reason but I have no reason to expect it to fail - everything passed previously in d90d2c4

**Comment by brianv0** [comment]
@conda-forge/help-python can we merge this?

**Comment by brianv0** [comment]
@conda-forge/staged-recipes ready

File: recipes/tables-io/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+  sha256: 2b98e4edc3c2fb4ca2452c7e8e3267bdb96b0a690d9fcf79d5ece3b96f7c99ad
+
+build:
+  number: 0
+  script: {{ PYTHON }} -m pip install . --no-deps --ignore-installed --no-cache-dir -vvv
+  noarch: python
+
+requirements:
+  host:
+   - python
```
Comment (beckermr) [inline recipes/tables-io/meta.yaml:20]:
```suggestion
   - python >=3.6
```

File: recipes/tables-io/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+   - pip
+   - setuptools_scm
+
+  run:
+    - python >=3.6
+    - numpy >1.21
```
Comment (beckermr) [inline recipes/tables-io/meta.yaml:26]: This requirement is really harsh. Does this package really need a numpy that is nearly bleeding edge?

File: recipes/tables-io/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,49 @@
+{% set name = "tables-io" %}
+{% set version = "0.0" %}
```
Comment (beckermr) [inline recipes/tables-io/meta.yaml:2]: I guess this is a real version, but it is rather odd-looking.

Comment (beckermr) [reply recipes/tables-io/meta.yaml:2]: Looks like 0.3 is upstream.

**Comment by beckermr** [comment]
@brianv0 back to you!

File: recipes/tables-io/meta.yaml (hunk 2 above)
Comment (beckermr) [reply recipes/tables-io/meta.yaml:26]: I read through the source code and all that is being called are things like `np.shape`, `np.ndim` etc. This pin is not needed and it is disruptive.

**Comment by brianv0** [comment]
Should be all good now, version was updated to 0.3

File: recipes/tables-io/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+{% set version = "0.3" %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  fn: {{ name }}-{{ version }}.tar.gz
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name.replace("-", "_") }}-{{ version }}.tar.gz
+  sha256: c7cfceaf5745160cf21d295751a22afd5839d93bb0bbead9090fc4922e1b85cf
```
Comment (beckermr) [inline recipes/tables-io/meta.yaml:11]:
```suggestion
  sha256: f526a3f2d917144ebe849dd059fae58aa05fa4f623abf9b855e220feba2454f6
```

**Pull request merged**